### DISCUSSION TITLE
Receiving "ValueError: cannot convert float NaN to integer" even when there are no NaNs in dataset.

### DISCUSSION THREAD

**Discussion by PratikSaha198** [TDAmeritrade/stumpy#227] [Q&A]
Receiving a ValueError: cannot convert float NaN to integer
on stump function  :
 --> 384     T_A, M_T, Σ_T = core.preprocess(T_A, m)
on preprocess function :
--> 722     T[np.isinf(T)] = np.nan

Even after rigorous checking of the dataset being fed into stumpy no nans where present or where found.

Python exception of preference in raising an exception to returning NaNs to things like sqrt(-1) and log(0.0) maybe causing the problem.

**Comment by seanlaw** [comment]
@PratikSaha198 Thank you for reporting this. Can you please share the inputs that you used that caused this error? This way, I can try to reproduce it on my end.

Also, can you tell me what version of STUMPY you are using. You can obtain this information with:

```
import stumpy

print(stumpy.__version__)
```

**Comment by imightbemary** [comment]
I'm experiencing the same error. I'm running version `1.4.0` and using Google Colab, which seems to be running Python 3.6.

I was working through the [STUMPY Basics tutorial](https://stumpy.readthedocs.io/en/latest/Tutorial_STUMPY_Basics.html) and encountered it when plugging in my own data. The actual code that threw the error is below.

```
m = 7
mp = stumpy.stump(cv['activity_7_day'], m)
```

The values in `cv['activity_7_day']` are as follows:

`array([258260,   6653, 137946, 889899, 136966, 229485, 167416, 252181,
       171082, 118669, 120038, 245277, 254121, 257795, 860418, 723360,
       188521, 131110,  84453, 303158, 771485, 163167,     47, 144024,
       797790, 156005, 373541,  41461, 651198,     21, 132940, 175323,
       470990,  83443,  32586, 194041, 261344, 158501, 342588, 223815,
       100990, 117814, 190712,  21957, 139347, 163310, 895826, 155270,
       231275, 106814,     18, 667925, 136602, 204493,  93868, 202681,
           33, 615518, 321356, 169991, 391335,  87394,  96485, 856649,
       220238, 256449,     33, 633095,  93813, 267410, 113307, 141562,
       154178, 503573, 197324, 181852, 489692, 104830, 775493, 286931,
       749440,  86396,  98444,     33, 601164,    336, 131347,  83435,
       872681, 874057, 204605,  86129, 212277, 277045,  91934,  88300,
       181280,      1,  94320,  13445, 105677, 242122, 408453, 190622,
       158664,  82371,  74115,  98629, 100011, 162429, 371517, 214390,
       207858, 134729, 112330, 417235, 262716,  79642, 588166,  84985,
       387164, 456257,  77129, 190514,     20,  94564, 716508,   2134,
        86430, 141321, 297542, 166135, 204133,  60064,  94599,  51495,
        89497,  92697, 200783, 358249, 137520, 147569,  85787, 397619,
       161909, 197019,  70385, 351880, 185797, 831303, 134197, 165784,
       200625, 431175, 148300, 103597, 140747, 128307, 193732, 219020,
           12, 516101, 243366,  96691,     15, 571045, 250792, 120512,
       877665,  91195, 864040, 275539, 177054, 475810, 194191, 409380,
       164666, 230069, 192619, 196431, 261759, 164920, 140678,     18,
       673890, 200701, 273413, 205959, 106502, 665024,     15, 592866,
       289106, 829299, 425963, 137943, 444807,     12, 542124,  86292,
        87522, 432834, 117613, 194417, 142726, 330039, 221258, 222511,
       704484, 129671, 314961,     21, 665039,  69117, 206059,  70849,
       195808, 142503,  65182, 166813, 426650, 512503,  82222, 258940,
       126476, 173205, 115256, 393722, 261470, 145937, 138997, 324592,
       281707, 191608])`

**Comment by seanlaw** [comment]
Thanks for the extra info, @imightbemary and I hope you are well! Let me take a look and see if I can reproduce it on my end. In the meantime, can you tell me what the `dtype` is on `cv['activity_7_day']`?  If it's some sort of `int`, can you try casting it `astype(float)` and try again?

**Comment by seanlaw** [comment]
Ahhh, indeed, STUMPY is failing to detect that the `dtype` is of type `int`. We usually have an internal function that checks for this but somehow it is failing to detect this and warn the user. For now, you should be able to overcome this with

```
m = 7
mp = stumpy.stump(cv['activity_7_day'].astype(float), m)
```

Sorry for the inconvenience/poor built-in error. I'll put a warning for this that will let the user know that only `float` is acceptable. This will be included in the upcoming v1.5.0 release (some time this week)

**Comment by imightbemary** [comment]
No worries--that did the trick for me! Thanks for the quick response, and hope you're doing well!

**Comment by seanlaw** [comment]
Awesome! Since you are using Colab, you may also be interested in leveraging the much faster `gpu_stump`. Here's [Colab notebook](http://tiny.cc/stumpy-colab) that walks you through it.

**Comment by imightbemary** [comment]
Perfect. I'll take a look!
